thrust/detail/preprocessor.h

```diff
@@ -1172,7 +1174,7 @@
   #define THRUST_CURRENT_FUNCTION __FUNC__
 #elif defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901)
   #define THRUST_CURRENT_FUNCTION __func__
-#elif defined(__cplusplus) && (__cplusplus >= 201103)
+#elif THRUST_CPP_DIALECT >= 2011
```

Comment:
This header by design includes nothing and is supposed to be self-contained (so that it can be used in other config code/stuff that is early in the dependency chain.

We don't need MSVC logic for `__cplusplus` here: MSVC has the `__FUNCSIG__` extension, which is checked for earlier in the #`ifdef` logic here.

So let's remove all the changes from this file, including the addition of the `#include`.